core/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -204,10 +204,13 @@ public String getLocalFilename(int blockIndex) throws IOException {
    * @throws IOException
    */
   public List<String> getLocationHosts() throws IOException {
+    List<String> ret = new ArrayList<String>();
```

Comment:
I think that is the current behavior, but this function is only used in TFShell which does not guard against null. I think returning an empty list is better than null since I regard null as undefined (as in the blocks could be somewhere but we do not know yet, whereas in this situation we do know there are no blocks).